### PR TITLE
Add Glm4MoeForCausalLM Support

### PR DESCRIPTION
**Pull request by quic-shagun**
This PR adds support for [zai-org/GLM-4.5-Air](https://huggingface.co/zai-org/GLM-4.5-Air) model. 
Open source MoE model with performance and accuracy better than many closed source models:

image

### CODE REVIEW

**Comment by quic-rishinr** [comment]
@shagsood do we have approval for this model? also do add this model under validated model list

**Review by vbaddi — CHANGES_REQUESTED** [review]

File: QEfficient/transformers/models/glm4_moe/modeling_glm4_moe.py [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+
+        attention_interface: Callable = eager_attention_forward
+
+        attn_output, attn_weights = attention_interface(
+            self,
+            query_states,
+            key_states,
+            value_states,
+            attention_mask,
+            dropout=0.0 if not self.training else self.attention_dropout,
```
Comment (asmigosw) [inline QEfficient/transformers/models/glm4_moe/modeling_glm4_moe.py]: Can we remove this dropout, since we are not using it in eager_attention_forward

**Review by asmigosw — CHANGES_REQUESTED** [review]

**Review by vbaddi — CHANGES_REQUESTED** [review]

**Comment by anujgupt-github** [comment]
@shagsood can you rebase and bring this to main branch?

**Comment by quic-rishinr** [comment]
@asmigosw @mamtsing please review the PR

**Comment by quic-rishinr** [comment]
@ochougul please fix the Lint DCO and other failures

**Comment by quic-hemagnih** [comment]
@ochougul  Can you please fix the LINT errors

**Comment by quic-rishinr** [comment]
New PR is raised with the following features. Will close this PR and will proceed with PR #988 
•	GLM4-MOE prefill/decode path
•	Chunked prefill MoE path with packed expert dispatch
•	KV-blocked attention path, headpar_offline to be default for all blocking combinations.
•	Disaggregated prefill/decode serving example
•	ONNX subfunction export for decode and prefill
